ci/cudf_pandas_scripts/pandas-tests/run.sh

```diff
@@ -17,8 +17,20 @@ RAPIDS_PY_WHEEL_NAME="cudf_${RAPIDS_PY_CUDA_SUFFIX}" rapids-download-wheels-from
 RAPIDS_PY_WHEEL_NAME="libcudf_${RAPIDS_PY_CUDA_SUFFIX}" rapids-download-wheels-from-s3 cpp ./dist
 RAPIDS_PY_WHEEL_NAME="pylibcudf_${RAPIDS_PY_CUDA_SUFFIX}" rapids-download-wheels-from-s3 python ./dist
 
+echo "" > ./constraints.txt
+if [[ $RAPIDS_DEPENDENCIES == "oldest" ]]; then
+    # `test_python` constraints are for `[test]` not `[cudf-pandas-tests]`
+    rapids-dependency-file-generator \
+        --output requirements \
+        --file-key test_python \
+        --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION};dependencies=${RAPIDS_DEPENDENCIES}" \
+    | tee ./constraints.txt
+fi
```

Comment:
Do we think this is going to be a common pattern that's worth extracting into a gha-tool `generate-oldest-constraints $file_key` or similar? @jameslamb WDYT? It occurs many times in this PR and I assume we'll do something similar in other repos.

Reply:
Yeah I think that's worth doing!

The only part that will vary from repo-to-repo is `--file-key`, everything else here is mechanical and a great candidate for a `gha-tools` script. I could put one up in a day or two, unless you or @seberg want to do it.

Reply:
Would suggest one of the two of you pick this up. Sebastian is out for a bit

Reply:
If you could tackle this that would be great @jameslamb. No rush. @mroeschke if you get everything else here finalized and have it ready to merge, I'm fine merging this PR with this code in and refactoring in a follow-up when James has the gha-tool ready.

Reply:
Put up a proposal in https://github.com/rapidsai/gha-tools/pull/114